lib/main.dart

```diff
@@ -1,18 +1,39 @@
+import 'package:bloc/bloc.dart';
 import 'package:dynamic_theme/dynamic_theme.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:hydrated_bloc/hydrated_bloc.dart';
 import 'package:provider/provider.dart';
 import 'package:pub_client/pub_client.dart';
 import 'package:tavern/screens/bloc.dart';
 import 'package:tavern/screens/home/home.dart';
-import 'package:tavern/screens/package_details_page.dart';
+import 'package:tavern/screens/package_details/package_details_bloc.dart';
+import 'package:tavern/screens/package_details/package_details_page.dart';
 import 'package:tavern/screens/search_screen.dart';
 import 'package:tavern/screens/settings/settings_screen.dart';
 import 'package:tavern/src/pub_colors.dart';
 
-void main() {
-  runApp(PubDevClientApp());
+Future main() async {
+  BlocSupervisor.delegate = await HydratedBlocDelegate.build();
+  PubHtmlParsingClient _htmlParsingClient;
+
+  _htmlParsingClient = PubHtmlParsingClient();
+  runApp(MultiBlocProvider(
+    providers: [
+      BlocProvider<HomeBloc>(
+        builder: (BuildContext context) => HomeBloc(client: _htmlParsingClient),
+      ),
+      BlocProvider<SettingsBloc>(
+        builder: (BuildContext context) => SettingsBloc(),
+      ),
+      BlocProvider<PackageDetailsBloc>(
+        builder: (BuildContext context) =>
+            PackageDetailsBloc(client: _htmlParsingClient),
+      ),
+    ],
+    child: PubDevClientApp(),
+  ));
 }
 
 class PubDevClientApp extends StatefulWidget {
```

Comment:
can we just make this a `StatelessWidget`?

Reply:
Done now, good eye, thanks